swagger-templates/src/main/java/com/okta/swagger/codegen/AbstractOktaJavaClientCodegen.java

```diff
@@ -151,14 +152,12 @@ protected void buildTopLevelResourceList(OpenAPI openAPI) {
 
         // find any children of these resources
         openAPI.getComponents().getSchemas().forEach((name, model) -> {
-            if(model.getExtensions() != null && model.getExtensions().containsKey("x-okta-parent")) {
-                String parent = (String) model.getExtensions().get("x-okta-parent");
-                if (parent != null) {
-                    parent = parent.replaceAll(".*/", "");
+            String parent = getParentModelRef(model);
+            if (parent != null) {
+                parent = parent.replaceAll(".*/", "");
 
-                    if (resources.contains(parent)) {
-                        resources.add(parent);
-                    }
+                if (resources.contains(parent)) {
+                    resources.add(name);
                 }
             }
```

Comment:
Two fixes here: 

1. Use `allOf` to find the parent since we are removing `x-okta-parent`
2. Fix bug that just adds the parent twice